shell/edit/provisioning.cattle.io.cluster/tabs/Advanced.vue

```diff
@@ -128,15 +164,18 @@ export default {
         :title="t('cluster.advanced.argInfo.machineSelector.kubeSchedulerTitle')"
       />
     </template>
-    <template v-if="agentArgs['protect-kernel-defaults']">
+    <template v-if="agentArgs && agentArgs['protect-kernel-defaults']">
```

Comment:
Yeah, this should be: `v-if="protectKernelDefaults !== undefined"`

Reply:
`agentArgs` is a sort of descriptor of what args can the user set up on his clusters; `protectKernelDefaults` is the value itself, that cannot be undefined since, with current changes, it should has a fallback on `machineGlobalConfig` if machineSelectorConfig is null.